### ISSUE TITLE
Problems routing traffic when multiple NICs

### ISSUE THREAD

**Issue by nsoubelet** [rancher/k3s#1131]
**Version:**
k3s version v1.0.0 (18bd921c)
INSTALL_K3S_EXEC="--no-deploy=traefik -i 10.xx.xx.xx --node-external-ip 10.xx.xx.xx"

**Describe the bug**
Since all my nodes have 2 nics, eth0 for internet (public ip) and eth1 for intranet, when default gateway is set to eth0 **cluster IPs** can't be resolved and when default is eth1 **public IPs** can't be resolved instead.
I searched extensively on net getting nothing helpful. Can't believe I am the only one facing this.
Just wanted to know whether someone has a similar setup or it is the intended functionality and nothing can be done.

**To Reproduce**
One needs at least 1 master and 1 node. Both with 2 NICs, one with a public ip while the other private.
Then deploy a pod on the node and try reach a public/private destination depending on which adapter is the default.

**Expected behavior**
As in the host machine, route the packages based on the IP. In the host I can live with those NICs and reach internal and external destinations without a problem.

**Actual behavior**
public/private destinations unreachable depending on which adapter is the default.

**Comment by erikwilson** [comment]
Does the `--flannel-iface` flag help?

**Comment by nsoubelet** [comment]
Hello @erikwilson 
Many thanks for your prompt answer. You are right, that was the solution. In fact, you already answered that to other user months ago and I tried days ago but with no luck because I made a mistake.

Just to elaborate for others looking for this similar issue, in case of the flag `--flannel-iface` is activated and a particular adapter specified, **all nodes and masters** should be pointed to the same ethx network, that is, if I have an eth1 with an internal ip and, for example, in other host the internal ip is in eth0, flannel must be pointing to those adapters just to let flannel communicate only via internal network.
I know that sounds a bit stupid, but might save a couple of hours of research. :)

**Comment by nsoubelet** [comment]
Hello @erikwilson 
Many thanks for your prompt answer. You are right, that was the solution. In fact, you already answered that to other user months ago and I tried days ago but with no luck because I made a mistake.

Just to elaborate for others looking for this similar issue, in case of the flag `--flannel-iface` is activated and a particular adapter specified, **all nodes and masters** should be pointed to the same ethx network, that is, if I have an eth1 with an internal ip and, for example, in other host the internal ip is in eth0, flannel must be pointing to those adapters just to let flannel communicate only via internal network.
I know that sounds a bit stupid, but might save a couple of hours of research. :)
